File: Shell/Modules/make.cc
#include "make.hh"

using namespace Fract::Shell::Modules;
using namespace Fract::Utilities;
using namespace Fract::Parser;

void
make::process(std::string cmd) {
  if(cmd == "")
  { std::cout << "This module cannot only be used!" << std::endl;
    return;
  }
  cmd += !string::ends_with(cmd, fract_extension) ? fract_extension : "";
  if(!file_system::exist_file(cmd))
  { std::cout << "The Fract file is not exists: " << cmd << std::endl;
    return;
  }
  parser entry(cmd, type_entry_file);
}

bool
make::check(std::string value) {
  if(string::ends_with(value, fract_extension)){
    return true;
  }
  else {
    value += fract_extension;
  }
  
  if(file_system::exist_file(value)) {
    return true;
  }
  return false;
}

File: shell/modules/make/make.go
package make

import (
	"fmt"
	"strings"

	"../../../fract"
	"../../../interpreter"
	"../../../utilities/fs"
)

// Process Process command in module.
// command Command to process.
func Process(command string) {
	if command == "" {
		fmt.Println("This module cannot only be used!")
		return
	}
	if !strings.HasSuffix(command, fract.FractExtension) {
		command += fract.FractExtension
	}
	if !fs.ExistFile(command) {
		fmt.Println("The Fract file is not exists: " + command)
		return
	}

	var preter = interpreter.New(command, fract.TypeEntryFile)
	preter.Interpret()
}

// Check Check invalid state of value.
// value Value to check.
func Check(value string) bool {
	if strings.HasSuffix(value, fract.FractExtension) {
		return true
	}
	value += ".fract"
	return fs.ExistFile(value)
}
